src/core/geometry/qgsnurbsutils.cpp

```diff
@@ -0,0 +1,243 @@
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsnurbsutils.h"
+
+#include "qgscompoundcurve.h"
+#include "qgscurvepolygon.h"
+#include "qgsgeometrycollection.h"
+#include "qgsnurbscurve.h"
+#include "qgsvertexid.h"
+
+bool QgsNurbsUtils::containsNurbsCurve( const QgsAbstractGeometry *geom )
+{
+  if ( !geom )
+    return false;
+
+  if ( qgsgeometry_cast<const QgsNurbsCurve *>( geom ) )
+    return true;
+
+  if ( const QgsGeometryCollection *gc = qgsgeometry_cast<const QgsGeometryCollection *>( geom ) )
+  {
+    for ( int i = 0; i < gc->numGeometries(); ++i )
+    {
+      if ( containsNurbsCurve( gc->geometryN( i ) ) )
+        return true;
+    }
+  }
+
+  if ( const QgsCurvePolygon *cp = qgsgeometry_cast<const QgsCurvePolygon *>( geom ) )
+  {
+    if ( containsNurbsCurve( cp->exteriorRing() ) )
+      return true;
+    for ( int i = 0; i < cp->numInteriorRings(); ++i )
+    {
+      if ( containsNurbsCurve( cp->interiorRing( i ) ) )
+        return true;
+    }
+  }
+
+  if ( const QgsCompoundCurve *cc = qgsgeometry_cast<const QgsCompoundCurve *>( geom ) )
+  {
+    for ( int i = 0; i < cc->nCurves(); ++i )
+    {
+      if ( containsNurbsCurve( cc->curveAt( i ) ) )
+        return true;
+    }
+  }
+
+  return false;
+}
+
+const QgsNurbsCurve *QgsNurbsUtils::extractNurbsCurve( const QgsAbstractGeometry *geom )
+{
+  if ( !geom )
+    return nullptr;
+
+  if ( const QgsNurbsCurve *nurbs = qgsgeometry_cast<const QgsNurbsCurve *>( geom ) )
+    return nurbs;
+
+  if ( const QgsGeometryCollection *gc = qgsgeometry_cast<const QgsGeometryCollection *>( geom ) )
+  {
+    for ( int i = 0; i < gc->numGeometries(); ++i )
+    {
+      const QgsNurbsCurve *nurbs = extractNurbsCurve( gc->geometryN( i ) );
+      if ( nurbs )
+        return nurbs;
+    }
+  }
+
+  if ( const QgsCurvePolygon *cp = qgsgeometry_cast<const QgsCurvePolygon *>( geom ) )
+  {
+    const QgsNurbsCurve *nurbs = extractNurbsCurve( cp->exteriorRing() );
+    if ( nurbs )
+      return nurbs;
+    for ( int i = 0; i < cp->numInteriorRings(); ++i )
+    {
+      nurbs = extractNurbsCurve( cp->interiorRing( i ) );
+      if ( nurbs )
+        return nurbs;
+    }
+  }
+
+  if ( const QgsCompoundCurve *cc = qgsgeometry_cast<const QgsCompoundCurve *>( geom ) )
+  {
+    for ( int i = 0; i < cc->nCurves(); ++i )
+    {
+      const QgsNurbsCurve *nurbs = extractNurbsCurve( cc->curveAt( i ) );
+      if ( nurbs )
+        return nurbs;
+    }
+  }
+
+  return nullptr;
+}
+
+bool QgsNurbsUtils::isPolyBezier( const QgsNurbsCurve *nurbs )
```

Comment:
is there a particular reason for not making isPolyBezier() a method of QgsNurbsCurve itself ?